src/execution/execute.js

```diff
@@ -322,18 +332,24 @@ function executeFields(
   exeContext: ExecutionContext,
   parentType: GraphQLObjectType,
   sourceValue: mixed,
+  exePath: Array<string | number>,
   fields: {[key: string]: Array<Field>}
 ): Object {
   let containsPromise = false;
 
   const finalResults = Object.keys(fields).reduce(
     (results, responseName) => {
       const fieldASTs = fields[responseName];
+
+      const childExePath = exePath.slice();
+      childExePath.push(responseName);
```

Comment:
Just a minor detail that the changes to the body of this function should match `executeFieldsSerially` as much as possible.